addons/koji/common/src/main/java/org/commonjava/indy/koji/inject/KojiProjectRefStringKey2StringMapper.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.koji.inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.commonjava.indy.koji.content.KojiMavenMetadataProvider;
+import org.commonjava.maven.atlas.ident.ref.SimpleProjectRef;
+import org.infinispan.persistence.keymappers.TwoWayKey2StringMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KojiProjectRefStringKey2StringMapper
+        implements TwoWayKey2StringMapper
+{
+    private final Logger logger = LoggerFactory.getLogger( this.getClass() );
+    private static final String SPLITTER = ":";
+    @Override
+    public Object getKeyMapping( String stringKey )
+    {
+        String[] parts = stringKey.split( SPLITTER );
```

Comment:
I think we have something like this for JSON parsing already...I wonder if we can reuse the logic:

* https://github.com/Commonjava/atlas/blob/master/bindings/jackson-identities/src/main/java/org/commonjava/maven/atlas/ident/jackson/ProjectRefDeserializer.java
* https://github.com/Commonjava/atlas/blob/master/bindings/jackson-identities/src/main/java/org/commonjava/maven/atlas/ident/jackson/SerializerIdentityUtils.java

Really, this amounts to calling `SimpleProjectRef.parse(..)`

In fact, you could literally make use of those Serializer / Deserializer classes by directly instantiating them...that would work fine too, and would align this work with that for any future bug fixes...wdyt?